Docs: add delegation use-case and clean up existing use-cases (#3138)
input-output-hk/cardano-wallet: docs/user-guide/common-use-cases/delegation.md

---
order: 6
title: Delegation
---

## Pre-requisites

 - [[how-to-start-wallet-server]]
 - [[how-to-create-a-wallet]]
 - In order to be able to send transactions we need funds on the wallet. In case of [Testnet](https://testnets.cardano.org/en/testnets/cardano/overview/) we can request tADA from the [faucet](https://testnets.cardano.org/en/testnets/cardano/tools/faucet/).

## Overview

> Delegation is the process by which ada holders delegate the stake associated with their ada to a stake pool. Cardano wallet allows listing all stake pools that are operating on the blockchain and then "joining" or "quitting" them via special delegation transaction.

Reviewer: ```suggestion
Delegation is the process by which ada holders delegate the stake associated with their ada to a stake pool. Cardano wallet allows listing all stake pools that are operating on the blockchain and then "join" or "quit" them via special delegation transaction.
```

> Delegation is supported only for **Shelley** wallets.

Reviewer: will be also for multisig - just the note to add later or now

## Listing stake pools

> Before joining any stake pool we can first list all available stake pools that are operating on our blockchain. Stake pools are ordered by `non_myopic_member_rewards` which makes pools which will produce potentially the best rewards in the future higher in the ranking. The ordering can also depend on the `?stake` query parameter which says how much stake we want to delegate.

Reviewer: ```suggestion
Before joining any stake pool we can first list all available stake pools that are operating on our blockchain. Stake pools are ordered by `non_myopic_member_rewards` which gives higher ranking and hence favors the pools potentially producing the best rewards in the future. The ordering could be influenced by the `?stake` query parameter which says how much stake we want to delegate.
```

[`GET /stake-pools`](https://input-output-hk.github.io/cardano-wallet/api/edge/#operation/listStakePools)

```
$ curl -X GET http://localhost:8090/v2/stake-pools?stake=1000000000
```

## Joining stake pool

> Once we select a stake pool we can "join" it. This operation will "virtually" add our wallet balance to the stake of this particular stake pool. Joining a pool for the first time will incur `fee` and `deposit` required for registering our stake key on the blockchain. This deposit will be returned to us if we choose to quit stake pool all together.

Reviewer: ```suggestion
Once we select a stake pool we can "join" it. This operation will "virtually" add our wallet balance to the stake of this particular stake pool. Joining a pool for the first time will incur `fee` and `deposit` required for registering our stake key on the blockchain. This deposit will be returned to us if we quit stake pool all together.
```

> > :information_source: The amount of the deposit is controlled by Cardano network parameter. It is currently 2₳ on `mainnet` and `testnet`. Deposit is taken only once, when joining stake pool for the first time.

Reviewer: ```suggestion
> :information_source: The amount of the deposit is controlled by Cardano network parameter. It is currently 2₳ on `mainnet` and `testnet`. Deposit is taken only once, when joining stake pool for the first time. Rejoining another stake pool does not incur another deposit.
```

 - Construct: [`POST /wallets/{walletId}/transactions-construct`](https://input-output-hk.github.io/cardano-wallet/api/edge/#operation/constructTransaction)
 - Sign: [`POST /wallets/{walletId}/transactions-sign`](https://input-output-hk.github.io/cardano-wallet/api/edge/#operation/signTransaction)
 - Submit: [`POST /wallets/{walletId}/transactions-submit`](https://input-output-hk.github.io/cardano-wallet/api/edge/#operation/submitTransaction)

Exemplary construct delegation request may look as follows:
```
$ curl -X POST http://localhost:8090/v2/wallets/1b0aa24994b4181e79116c131510f2abf6cdaa4f/transactions-construct \  
-d '{"delegations":
     [{"join":{"pool":"pool1mgjlw24rg8sp4vrzctqxtf2nn29rjhtkq2kdzvf4tcjd5pl547k",
               "stake_key_index":"0H"}}]}' \  
-H "Content-Type: application/json"
```

Refer to [[how-to-make-a-transaction]] for details on signing and submitting it to the network.

> :information_source: Note that information about `deposit_taken` and `deposit_returned` is available in the wallet transaction history. Refer to [[how-to-make-a-transaction]].

## Joining another stake pool

> Joining another stake pool doesn't differ from the previous process at all. The only difference is that, behind the scenes, `deposit` is not taken from the wallet, as it was already taken when joining any stake pool for the first time.

Reviewer: ```suggestion
Joining another stake pool doesn't differ from the previous process at all. The only difference is that, behind the scenes, `deposit` is not taken from the wallet, as it was already taken when joining for the first time.
```

> :information_source: Note that information about `withdrawals` is also available in the wallet transaction history. Refer to [[how-to-make-a-transaction]].

## Quitting stake pool

At any point we can decide to quit delegation all together. Quitting stake pool will cause de-registration of the stake key associated to our wallet and our deposit will be returned. After that we're not longer going to receive rewards.

Quitting can be done in old transaction workflow:

[`DELETE /stake-pools/*/wallets/{walletId}`](https://input-output-hk.github.io/cardano-wallet/api/edge/#operation/quitStakePool)

> :information_source: Note that all rewards will be withdrawn automatically upon quitting a stake pool.

or [new transaction workflow](https://input-output-hk.github.io/cardano-wallet/api/edge/#tag/Transactions-New), for instance:

Just quitting:
```
$ curl -X POST http://localhost:8090/v2/wallets/1b0aa24994b4181e79116c131510f2abf6cdaa4f/transactions-construct \  
-d '{"delegations":[{"quit":{"stake_key_index":"0H"}}]}' \  
-H "Content-Type: application/json"
```

> Quitting and withdrawing rewards in the same transaction:

Reviewer: ```suggestion
Quitting and withdrawing rewards in the same transaction. It should be noted that quitting can be realized only when all rewards are withdrawn:
```
